### PR TITLE
New package: maciak.whatismyip version 0.2.1

### PR DESCRIPTION
**Pull request by maciakl**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/118293)

### CODE REVIEW

**Comment by maciakl** [comment]
@microsoft-github-policy-service agree

**Comment by stephengillie** [comment]
Hi @maciakl,

This package didn't pass a Defender scan:
image

Please scan your systems. This might be a false positive, and we can rescan tomorrow.

**Comment by maciakl** [comment]
@stephengillie, thanks. I updated and re-uploaded the binary and re-scanned it with Defender and Malwarebytes. I get no detentions this time. Let me know if it passes on your end.

I also submitted the binary to Virustotal and it looks like it got 2 false positive heuristic detections from obscure vendors (probably on account of what the tool is doing: which is looking up the routable and local ip addresses of the host machine):

https://www.virustotal.com/gui/file/4a8aa5cbe01b8d3d904af89365a26949019a21e16bbe834f9b855223a2f1f96d

The tool is literally 63 lines of code with 3 dependencies (clap, reqwest and local-ip-address crates from rust.io). The source is published here: https://github.com/maciakl/whatismyip

**Comment by stephengillie** [comment]
Sometimes, the virus can be hidden in an icon or other artifact. Were any online icon creation services used?

**Comment by maciakl** [comment]
No, no icons or graphics.

It’s a very simple command line tool. When you run it it makes a single http get request to ipify.org service to identify the host machines external IP address, prints it to the terminal, then exits.

It looks like the ipify service has been blacklisted by some vendors due to some malware making requests to their various API’s in the past (tho the service itself appears to be legit and above the board):

https://github.com/rdegges/ipify-api/issues/2

I’m assuming this is what might be generating the false positives. I will try rebuilding using a different service and see if this  helps.

**Comment by maciakl** [comment]
On a lark, I wrote an even simpler version of this tool on an unrelated machine (on a different network) in go instead of rust. It was literally 10 lines and no 3rd party dependencies this time. I also used different web service (my-ip.io instead of ipify this time).

Compiled it, submitted to Virus Total, still get detections from the exact same vendors:

https://www.virustotal.com/gui/file/328aafed4790fcaaa0e3afa7bfdb4574ae79efdec5ee90a3c47bfb73086bc11f/detection

I think I can rule out an infected PC, or a shady dependency at this point. It looks like some AV's really don't like small windows executables that do DNS lookups and get requests to IP discovery services when launched.

At this point I'm not sure if anything can be done about this as IP discovery the entire point of this tiny little tool.

Let me know if you can't get a clean scan on your end. If it's a blocking issue I'll just close the pull request and deal with not having the tool installable via `winget`.

**Comment by stephengillie** [comment]
Hi @maciakl,

Sorry to forget this PR for so long.

Manual Validation ended with:
> 3:28:49 PM:  > 2023-11-01 15:28:48.983 [CLI ] Package hash verification failed. SHA256 in manifest 
[5c937601e915ac72083fc4ed3206ad9b2c5aeffebbaf8ca0f249bf495761e9d1] does not match download 
[ff861c987c345f4c5ba1c442238be4c09ee7c0efa93b9796844a099e5cce5228]

File: manifests/m/maciak/whatismyip/0.2.1/maciak.whatismyip.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+PackageVersion: 0.2.1
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: whatismyip.exe
+  PortableCommandAlias: whatismyip
+Installers:
+- InstallerUrl: https://github.com/maciakl/whatismyip/releases/download/0.2.1/whatismyip_windows.zip
+  Architecture: x64
+  InstallerSha256: 5C937601E915AC72083FC4ED3206AD9B2C5AEFFEBBAF8CA0F249BF495761E9D1
```
Comment (stephengillie) [inline manifests/m/maciak/whatismyip/0.2.1/maciak.whatismyip.installer.yaml:14]:
```suggestion
  InstallerSha256: FF861C987C345F4C5BA1C442238BE4C09EE7C0EFA93B9796844A099E5CCE5228
```

**Pull request closed without merging**